README.md

```diff
@@ -218,6 +218,144 @@ OpenStack | N | N | Y | N | N
 oVirt | N | N | Y | N | N
 VMWare | N | N | Y | N | N
 
+# Short lived Credentials with AWS Secure Token Service
+### Overview
+We can also make Openshift use temporary credentials for different components with AWS Secure Token Service (STS). It enables an authentication flow allowing a component to assume an IAM Role resulting in short-lived credentials. It also automates requesting and refreshing of credentials using OpenID Connect Identity Provider. Openshift can sign ServiceAccount token trusted by AWS IAM which can be projected into a Pod and used for authentication. The following is a diagram showing how it works.
+
+![sts flow](docs/sts_flow.png)
+
+### Changes in the Credentials Secret with STS
+
+Without STS we use access key id and secret access key
+
+```yaml
+aws_access_key_id = <access_key_id>
+secret_access_key = <secret_access_key>
+```
+
+With STS we use role and web identity token
+
+```yaml
+role_name = arn:...:role/some-role-name
+web_identity_token_file = /path/to/token
+```
+The token is short lived for an hour after which it is refreshed.
+
+### Steps to install an OpenShift Cluster with STS
+
+1. Use [this](https://quay.io/repository/jdiaz/custom-release) custom OpenShift release image
+2. Extract the AWS Credentials Request objects from the above release image. Please use [4.6](https://openshift-release-artifacts.apps.ci.l2s4.p1.openshiftapps.com/4.6.0/) openshift client (oc)
+   ```
+   oc adm release extract --cloud=aws --credentials-requests quay.io/jdiaz/custom-release:sts > credreqs.yaml
+   ```
+3. Extract the OpenShift install binary from the release image
+   ```
+   oc adm release extract --command=openshift-install quay.io/jdiaz/custom-release:sts
+   ```
+4. Create an install-config.yaml
+   ```
+   ./openshift-install create install-config
+   ```
+5. Make sure that we install the cluster in Manual mode
+   ```
+   echo "credentialsMode: Manual" >> install-config.yaml
+   ``` 
+6. Create install manifests
+   ```
+   ./openshift-install create manifests   
+   ```
+7. Create AWS resources using [sts-preflight](https://github.com/sjenning/sts-preflight) tool. Below command will create the S3 bucket, upload the OIDC config into the bucket, set up an IAM Identity Provider that trusts that bucket, create IAM Roles for each AWS CredentialsRequest. It will also dump the files needed by the installer in the `_output` directory
```

Comment:
```suggestion
7. Create AWS resources using [sts-preflight](https://github.com/sjenning/sts-preflight) tool. Below command will generate public/private ServiceAccount signing keys, create the S3 bucket, upload the OIDC config into the bucket, set up an IAM Identity Provider that trusts that bucket, an IAM Role to be used for installation, and create IAM Roles for each AWS CredentialsRequest. It will also dump the files needed by the installer in the `_output` directory
```

Reply:
> an IAM Role to be used for installation

we should be NOTE this is role has Admin rights last i checked.